<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns:mv="http://macVmlSchemaUri" xmlns="http://www.w3.org/TR/REC-html40">

<head>
<meta name=Title content="">
<meta name=Keywords content="">
<meta http-equiv=Content-Type content="text/html; charset=macintosh">
<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 14">
<meta name=Originator content="Microsoft Word 14">
<link rel=File-List href="course_16S_files/filelist.xml">
<!--[if gte mso 9]><xml>
 <o:DocumentProperties>
  <o:Author>Brandi Cantarel</o:Author>
  <o:LastAuthor>admin</o:LastAuthor>
  <o:Revision>9</o:Revision>
  <o:TotalTime>160</o:TotalTime>
  <o:Created>2015-10-18T23:40:00Z</o:Created>
  <o:LastSaved>2016-10-19T00:41:00Z</o:LastSaved>
  <o:Pages>3</o:Pages>
  <o:Words>1950</o:Words>
  <o:Characters>11119</o:Characters>
  <o:Company>UMIGS</o:Company>
  <o:Lines>92</o:Lines>
  <o:Paragraphs>26</o:Paragraphs>
  <o:CharactersWithSpaces>13043</o:CharactersWithSpaces>
  <o:Version>14.0</o:Version>
 </o:DocumentProperties>
 <o:OfficeDocumentSettings>
  <o:AllowPNG/>
 </o:OfficeDocumentSettings>
</xml><![endif]-->
<link rel=themeData href="course_16S_files/themedata.xml">
<!--[if gte mso 9]><xml>
 <w:WordDocument>
  <w:Zoom>125</w:Zoom>
  <w:SpellingState>Clean</w:SpellingState>
  <w:GrammarState>Clean</w:GrammarState>
  <w:TrackMoves/>
  <w:TrackFormatting/>
  <w:PunctuationKerning/>
  <w:DrawingGridHorizontalSpacing>18 pt</w:DrawingGridHorizontalSpacing>
  <w:DrawingGridVerticalSpacing>18 pt</w:DrawingGridVerticalSpacing>
  <w:DisplayHorizontalDrawingGridEvery>0</w:DisplayHorizontalDrawingGridEvery>
  <w:DisplayVerticalDrawingGridEvery>0</w:DisplayVerticalDrawingGridEvery>
  <w:ValidateAgainstSchemas/>
  <w:SaveIfXMLInvalid>false</w:SaveIfXMLInvalid>
  <w:IgnoreMixedContent>false</w:IgnoreMixedContent>
  <w:AlwaysShowPlaceholderText>false</w:AlwaysShowPlaceholderText>
  <w:DoNotPromoteQF/>
  <w:LidThemeOther>EN-US</w:LidThemeOther>
  <w:LidThemeAsian>JA</w:LidThemeAsian>
  <w:LidThemeComplexScript>X-NONE</w:LidThemeComplexScript>
  <w:Compatibility>
   <w:BreakWrappedTables/>
   <w:SnapToGridInCell/>
   <w:WrapTextWithPunct/>
   <w:UseAsianBreakRules/>
   <w:DontGrowAutofit/>
   <w:SplitPgBreakAndParaMark/>
   <w:EnableOpenTypeKerning/>
   <w:DontFlipMirrorIndents/>
   <w:OverrideTableStyleHps/>
   <w:UseFELayout/>
  </w:Compatibility>
  <m:mathPr>
   <m:mathFont m:val="Cambria Math"/>
   <m:brkBin m:val="before"/>
   <m:brkBinSub m:val="&#45;-"/>
   <m:smallFrac/>
   <m:dispDef/>
   <m:lMargin m:val="0"/>
   <m:rMargin m:val="0"/>
   <m:defJc m:val="centerGroup"/>
   <m:wrapRight/>
   <m:intLim m:val="subSup"/>
   <m:naryLim m:val="subSup"/>
  </m:mathPr></w:WordDocument>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <w:LatentStyles DefLockedState="false" DefUnhideWhenUsed="true"
  DefSemiHidden="true" DefQFormat="false" DefPriority="99"
  LatentStyleCount="276">
  <w:LsdException Locked="false" Priority="0" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Normal"/>
  <w:LsdException Locked="false" Priority="9" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="heading 1"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 2"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 3"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 4"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 5"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 6"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 7"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 8"/>
  <w:LsdException Locked="false" Priority="9" QFormat="true" Name="heading 9"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 1"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 2"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 3"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 4"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 5"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 6"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 7"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 8"/>
  <w:LsdException Locked="false" Priority="39" Name="toc 9"/>
  <w:LsdException Locked="false" Priority="35" QFormat="true" Name="caption"/>
  <w:LsdException Locked="false" Priority="10" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Title"/>
  <w:LsdException Locked="false" Priority="1" Name="Default Paragraph Font"/>
  <w:LsdException Locked="false" Priority="11" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Subtitle"/>
  <w:LsdException Locked="false" Priority="22" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Strong"/>
  <w:LsdException Locked="false" Priority="20" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Emphasis"/>
  <w:LsdException Locked="false" Priority="59" SemiHidden="false"
   UnhideWhenUsed="false" Name="Table Grid"/>
  <w:LsdException Locked="false" UnhideWhenUsed="false" Name="Placeholder Text"/>
  <w:LsdException Locked="false" Priority="1" UnhideWhenUsed="false"
   QFormat="true" Name="No Spacing"/>
  <w:LsdException Locked="false" Priority="60" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Shading"/>
  <w:LsdException Locked="false" Priority="61" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light List"/>
  <w:LsdException Locked="false" Priority="62" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Grid"/>
  <w:LsdException Locked="false" Priority="63" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 1"/>
  <w:LsdException Locked="false" Priority="64" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 2"/>
  <w:LsdException Locked="false" Priority="65" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 1"/>
  <w:LsdException Locked="false" Priority="66" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 2"/>
  <w:LsdException Locked="false" Priority="67" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 1"/>
  <w:LsdException Locked="false" Priority="68" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 2"/>
  <w:LsdException Locked="false" Priority="69" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 3"/>
  <w:LsdException Locked="false" Priority="70" SemiHidden="false"
   UnhideWhenUsed="false" Name="Dark List"/>
  <w:LsdException Locked="false" Priority="71" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Shading"/>
  <w:LsdException Locked="false" Priority="72" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful List"/>
  <w:LsdException Locked="false" Priority="73" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Grid"/>
  <w:LsdException Locked="false" Priority="60" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Shading Accent 1"/>
  <w:LsdException Locked="false" Priority="61" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light List Accent 1"/>
  <w:LsdException Locked="false" Priority="62" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Grid Accent 1"/>
  <w:LsdException Locked="false" Priority="63" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 1 Accent 1"/>
  <w:LsdException Locked="false" Priority="64" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 2 Accent 1"/>
  <w:LsdException Locked="false" Priority="65" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 1 Accent 1"/>
  <w:LsdException Locked="false" UnhideWhenUsed="false" Name="Revision"/>
  <w:LsdException Locked="false" Priority="34" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="List Paragraph"/>
  <w:LsdException Locked="false" Priority="29" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Quote"/>
  <w:LsdException Locked="false" Priority="30" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Intense Quote"/>
  <w:LsdException Locked="false" Priority="66" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 2 Accent 1"/>
  <w:LsdException Locked="false" Priority="67" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 1 Accent 1"/>
  <w:LsdException Locked="false" Priority="68" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 2 Accent 1"/>
  <w:LsdException Locked="false" Priority="69" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 3 Accent 1"/>
  <w:LsdException Locked="false" Priority="70" SemiHidden="false"
   UnhideWhenUsed="false" Name="Dark List Accent 1"/>
  <w:LsdException Locked="false" Priority="71" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Shading Accent 1"/>
  <w:LsdException Locked="false" Priority="72" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful List Accent 1"/>
  <w:LsdException Locked="false" Priority="73" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Grid Accent 1"/>
  <w:LsdException Locked="false" Priority="60" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Shading Accent 2"/>
  <w:LsdException Locked="false" Priority="61" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light List Accent 2"/>
  <w:LsdException Locked="false" Priority="62" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Grid Accent 2"/>
  <w:LsdException Locked="false" Priority="63" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 1 Accent 2"/>
  <w:LsdException Locked="false" Priority="64" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 2 Accent 2"/>
  <w:LsdException Locked="false" Priority="65" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 1 Accent 2"/>
  <w:LsdException Locked="false" Priority="66" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 2 Accent 2"/>
  <w:LsdException Locked="false" Priority="67" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 1 Accent 2"/>
  <w:LsdException Locked="false" Priority="68" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 2 Accent 2"/>
  <w:LsdException Locked="false" Priority="69" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 3 Accent 2"/>
  <w:LsdException Locked="false" Priority="70" SemiHidden="false"
   UnhideWhenUsed="false" Name="Dark List Accent 2"/>
  <w:LsdException Locked="false" Priority="71" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Shading Accent 2"/>
  <w:LsdException Locked="false" Priority="72" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful List Accent 2"/>
  <w:LsdException Locked="false" Priority="73" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Grid Accent 2"/>
  <w:LsdException Locked="false" Priority="60" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Shading Accent 3"/>
  <w:LsdException Locked="false" Priority="61" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light List Accent 3"/>
  <w:LsdException Locked="false" Priority="62" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Grid Accent 3"/>
  <w:LsdException Locked="false" Priority="63" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 1 Accent 3"/>
  <w:LsdException Locked="false" Priority="64" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 2 Accent 3"/>
  <w:LsdException Locked="false" Priority="65" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 1 Accent 3"/>
  <w:LsdException Locked="false" Priority="66" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 2 Accent 3"/>
  <w:LsdException Locked="false" Priority="67" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 1 Accent 3"/>
  <w:LsdException Locked="false" Priority="68" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 2 Accent 3"/>
  <w:LsdException Locked="false" Priority="69" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 3 Accent 3"/>
  <w:LsdException Locked="false" Priority="70" SemiHidden="false"
   UnhideWhenUsed="false" Name="Dark List Accent 3"/>
  <w:LsdException Locked="false" Priority="71" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Shading Accent 3"/>
  <w:LsdException Locked="false" Priority="72" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful List Accent 3"/>
  <w:LsdException Locked="false" Priority="73" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Grid Accent 3"/>
  <w:LsdException Locked="false" Priority="60" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Shading Accent 4"/>
  <w:LsdException Locked="false" Priority="61" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light List Accent 4"/>
  <w:LsdException Locked="false" Priority="62" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Grid Accent 4"/>
  <w:LsdException Locked="false" Priority="63" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 1 Accent 4"/>
  <w:LsdException Locked="false" Priority="64" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 2 Accent 4"/>
  <w:LsdException Locked="false" Priority="65" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 1 Accent 4"/>
  <w:LsdException Locked="false" Priority="66" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 2 Accent 4"/>
  <w:LsdException Locked="false" Priority="67" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 1 Accent 4"/>
  <w:LsdException Locked="false" Priority="68" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 2 Accent 4"/>
  <w:LsdException Locked="false" Priority="69" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 3 Accent 4"/>
  <w:LsdException Locked="false" Priority="70" SemiHidden="false"
   UnhideWhenUsed="false" Name="Dark List Accent 4"/>
  <w:LsdException Locked="false" Priority="71" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Shading Accent 4"/>
  <w:LsdException Locked="false" Priority="72" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful List Accent 4"/>
  <w:LsdException Locked="false" Priority="73" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Grid Accent 4"/>
  <w:LsdException Locked="false" Priority="60" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Shading Accent 5"/>
  <w:LsdException Locked="false" Priority="61" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light List Accent 5"/>
  <w:LsdException Locked="false" Priority="62" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Grid Accent 5"/>
  <w:LsdException Locked="false" Priority="63" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 1 Accent 5"/>
  <w:LsdException Locked="false" Priority="64" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 2 Accent 5"/>
  <w:LsdException Locked="false" Priority="65" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 1 Accent 5"/>
  <w:LsdException Locked="false" Priority="66" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 2 Accent 5"/>
  <w:LsdException Locked="false" Priority="67" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 1 Accent 5"/>
  <w:LsdException Locked="false" Priority="68" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 2 Accent 5"/>
  <w:LsdException Locked="false" Priority="69" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 3 Accent 5"/>
  <w:LsdException Locked="false" Priority="70" SemiHidden="false"
   UnhideWhenUsed="false" Name="Dark List Accent 5"/>
  <w:LsdException Locked="false" Priority="71" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Shading Accent 5"/>
  <w:LsdException Locked="false" Priority="72" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful List Accent 5"/>
  <w:LsdException Locked="false" Priority="73" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Grid Accent 5"/>
  <w:LsdException Locked="false" Priority="60" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Shading Accent 6"/>
  <w:LsdException Locked="false" Priority="61" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light List Accent 6"/>
  <w:LsdException Locked="false" Priority="62" SemiHidden="false"
   UnhideWhenUsed="false" Name="Light Grid Accent 6"/>
  <w:LsdException Locked="false" Priority="63" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 1 Accent 6"/>
  <w:LsdException Locked="false" Priority="64" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Shading 2 Accent 6"/>
  <w:LsdException Locked="false" Priority="65" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 1 Accent 6"/>
  <w:LsdException Locked="false" Priority="66" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium List 2 Accent 6"/>
  <w:LsdException Locked="false" Priority="67" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 1 Accent 6"/>
  <w:LsdException Locked="false" Priority="68" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 2 Accent 6"/>
  <w:LsdException Locked="false" Priority="69" SemiHidden="false"
   UnhideWhenUsed="false" Name="Medium Grid 3 Accent 6"/>
  <w:LsdException Locked="false" Priority="70" SemiHidden="false"
   UnhideWhenUsed="false" Name="Dark List Accent 6"/>
  <w:LsdException Locked="false" Priority="71" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Shading Accent 6"/>
  <w:LsdException Locked="false" Priority="72" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful List Accent 6"/>
  <w:LsdException Locked="false" Priority="73" SemiHidden="false"
   UnhideWhenUsed="false" Name="Colorful Grid Accent 6"/>
  <w:LsdException Locked="false" Priority="19" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Subtle Emphasis"/>
  <w:LsdException Locked="false" Priority="21" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Intense Emphasis"/>
  <w:LsdException Locked="false" Priority="31" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Subtle Reference"/>
  <w:LsdException Locked="false" Priority="32" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Intense Reference"/>
  <w:LsdException Locked="false" Priority="33" SemiHidden="false"
   UnhideWhenUsed="false" QFormat="true" Name="Book Title"/>
  <w:LsdException Locked="false" Priority="37" Name="Bibliography"/>
  <w:LsdException Locked="false" Priority="39" QFormat="true" Name="TOC Heading"/>
 </w:LatentStyles>
</xml><![endif]-->
<style>
<!--
 /* Font Definitions */
@font-face
	{font-family:Times;
	panose-1:2 0 5 0 0 0 0 0 0 0;
	mso-font-charset:0;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:3 0 0 0 1 0;}
@font-face
	{font-family:"\FF2D\FF33 \660E\671D";
	mso-font-charset:78;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:1 134676480 16 0 131072 0;}
@font-face
	{font-family:"\FF2D\FF33 \660E\671D";
	mso-font-charset:78;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:1 134676480 16 0 131072 0;}
@font-face
	{font-family:Cambria;
	panose-1:2 4 5 3 5 4 6 3 2 4;
	mso-font-charset:0;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:-536870145 1073743103 0 0 415 0;}
@font-face
	{font-family:Georgia;
	panose-1:2 4 5 2 5 4 5 2 3 3;
	mso-font-charset:0;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:647 0 0 0 159 0;}
@font-face
	{font-family:Monaco;
	panose-1:2 0 5 0 0 0 0 0 0 0;
	mso-font-charset:0;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:3 0 0 0 1 0;}
 /* Style Definitions */
p.MsoNormal, li.MsoNormal, div.MsoNormal
	{mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	margin:0in;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:Cambria;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:"\FF2D\FF33 \660E\671D";
	mso-fareast-theme-font:minor-fareast;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-theme-font:minor-bidi;}
h2
	{mso-style-noshow:yes;
	mso-style-priority:9;
	mso-style-qformat:yes;
	mso-style-link:"Heading 2 Char";
	mso-margin-top-alt:auto;
	margin-right:0in;
	mso-margin-bottom-alt:auto;
	margin-left:0in;
	mso-pagination:widow-orphan;
	mso-outline-level:2;
	font-size:18.0pt;
	font-family:Times;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-theme-font:minor-bidi;
	font-weight:bold;}
a:link, span.MsoHyperlink
	{mso-style-noshow:yes;
	mso-style-priority:99;
	color:blue;
	mso-themecolor:hyperlink;
	text-decoration:underline;
	text-underline:single;}
a:visited, span.MsoHyperlinkFollowed
	{mso-style-noshow:yes;
	mso-style-priority:99;
	color:purple;
	mso-themecolor:followedhyperlink;
	text-decoration:underline;
	text-underline:single;}
p
	{mso-style-noshow:yes;
	mso-style-priority:99;
	mso-margin-top-alt:auto;
	margin-right:0in;
	mso-margin-bottom-alt:auto;
	margin-left:0in;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:Times;
	mso-fareast-font-family:"\FF2D\FF33 \660E\671D";
	mso-fareast-theme-font:minor-fareast;
	mso-bidi-font-family:"Times New Roman";}
pre
	{mso-style-noshow:yes;
	mso-style-priority:99;
	mso-style-link:"HTML Preformatted Char";
	margin:0in;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt;
	font-size:10.0pt;
	font-family:Courier;
	mso-fareast-font-family:"\FF2D\FF33 \660E\671D";
	mso-fareast-theme-font:minor-fareast;
	mso-bidi-font-family:Courier;}
p.MsoListParagraph, li.MsoListParagraph, div.MsoListParagraph
	{mso-style-noshow:yes;
	mso-style-priority:34;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	margin-top:0in;
	margin-right:0in;
	margin-bottom:0in;
	margin-left:.5in;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:Cambria;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:"\FF2D\FF33 \660E\671D";
	mso-fareast-theme-font:minor-fareast;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-theme-font:minor-bidi;}
p.MsoListParagraphCxSpFirst, li.MsoListParagraphCxSpFirst, div.MsoListParagraphCxSpFirst
	{mso-style-noshow:yes;
	mso-style-priority:34;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-type:export-only;
	margin-top:0in;
	margin-right:0in;
	margin-bottom:0in;
	margin-left:.5in;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:Cambria;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:"\FF2D\FF33 \660E\671D";
	mso-fareast-theme-font:minor-fareast;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-theme-font:minor-bidi;}
p.MsoListParagraphCxSpMiddle, li.MsoListParagraphCxSpMiddle, div.MsoListParagraphCxSpMiddle
	{mso-style-noshow:yes;
	mso-style-priority:34;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-type:export-only;
	margin-top:0in;
	margin-right:0in;
	margin-bottom:0in;
	margin-left:.5in;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:Cambria;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:"\FF2D\FF33 \660E\671D";
	mso-fareast-theme-font:minor-fareast;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-theme-font:minor-bidi;}
p.MsoListParagraphCxSpLast, li.MsoListParagraphCxSpLast, div.MsoListParagraphCxSpLast
	{mso-style-noshow:yes;
	mso-style-priority:34;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-type:export-only;
	margin-top:0in;
	margin-right:0in;
	margin-bottom:0in;
	margin-left:.5in;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:Cambria;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:"\FF2D\FF33 \660E\671D";
	mso-fareast-theme-font:minor-fareast;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-theme-font:minor-bidi;}
span.Heading2Char
	{mso-style-name:"Heading 2 Char";
	mso-style-noshow:yes;
	mso-style-priority:9;
	mso-style-unhide:no;
	mso-style-locked:yes;
	mso-style-link:"Heading 2";
	mso-ansi-font-size:18.0pt;
	mso-bidi-font-size:18.0pt;
	font-family:Times;
	mso-ascii-font-family:Times;
	mso-hansi-font-family:Times;
	font-weight:bold;}
span.HTMLPreformattedChar
	{mso-style-name:"HTML Preformatted Char";
	mso-style-noshow:yes;
	mso-style-priority:99;
	mso-style-unhide:no;
	mso-style-locked:yes;
	mso-style-link:"HTML Preformatted";
	mso-ansi-font-size:10.0pt;
	mso-bidi-font-size:10.0pt;
	font-family:Courier;
	mso-ascii-font-family:Courier;
	mso-hansi-font-family:Courier;
	mso-bidi-font-family:Courier;}
span.mw-headline
	{mso-style-name:mw-headline;
	mso-style-unhide:no;}
span.apple-converted-space
	{mso-style-name:apple-converted-space;
	mso-style-unhide:no;}
span.im
	{mso-style-name:im;
	mso-style-unhide:no;}
span.SpellE
	{mso-style-name:"";
	mso-spl-e:yes;}
span.GramE
	{mso-style-name:"";
	mso-gram-e:yes;}
.MsoChpDefault
	{mso-style-type:export-only;
	mso-default-props:yes;
	font-size:10.0pt;
	mso-ansi-font-size:10.0pt;
	mso-bidi-font-size:10.0pt;
	font-family:Cambria;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-fareast-font-family:"\FF2D\FF33 \660E\671D";
	mso-fareast-theme-font:minor-fareast;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-theme-font:minor-bidi;}
@page WordSection1
	{size:8.5in 11.0in;
	margin:1.0in 1.25in 1.0in 1.25in;
	mso-header-margin:.5in;
	mso-footer-margin:.5in;
	mso-paper-source:0;}
div.WordSection1
	{page:WordSection1;}
 /* List Definitions */
@list l0
	{mso-list-id:1453354841;
	mso-list-type:hybrid;
	mso-list-template-ids:474648398 67698705 67698713 67698715 67698703 67698713 67698715 67698703 67698713 67698715;}
@list l0:level1
	{mso-level-text:"%1\)";
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level2
	{mso-level-number-format:alpha-lower;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level3
	{mso-level-number-format:roman-lower;
	mso-level-tab-stop:none;
	mso-level-number-position:right;
	text-indent:-9.0pt;}
@list l0:level4
	{mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level5
	{mso-level-number-format:alpha-lower;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level6
	{mso-level-number-format:roman-lower;
	mso-level-tab-stop:none;
	mso-level-number-position:right;
	text-indent:-9.0pt;}
@list l0:level7
	{mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level8
	{mso-level-number-format:alpha-lower;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level9
	{mso-level-number-format:roman-lower;
	mso-level-tab-stop:none;
	mso-level-number-position:right;
	text-indent:-9.0pt;}
ol
	{margin-bottom:0in;}
ul
	{margin-bottom:0in;}
-->
</style>
<!--[if gte mso 10]>
<style>
 /* Style Definitions */
table.MsoNormalTable
	{mso-style-name:"Table Normal";
	mso-tstyle-rowband-size:0;
	mso-tstyle-colband-size:0;
	mso-style-noshow:yes;
	mso-style-priority:99;
	mso-style-parent:"";
	mso-padding-alt:0in 5.4pt 0in 5.4pt;
	mso-para-margin:0in;
	mso-para-margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:Cambria;
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;}
</style>
<![endif]--><!--[if gte mso 9]><xml>
 <o:shapedefaults v:ext="edit" spidmax="1027"/>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <o:shapelayout v:ext="edit">
  <o:idmap v:ext="edit" data="1"/>
 </o:shapelayout></xml><![endif]-->
</head>

<body lang=EN-US link=blue vlink=purple style='tab-interval:.5in'>

<div class=WordSection1>

<p class=MsoNormal><span style='font-family:Helvetica'>This exercise is adapted
from the <span class=SpellE>Mothur</span> <span class=SpellE>MiSeq</span> SOP<o:p></o:p></span></p>

<p class=MsoNormal><a href="http://www.mothur.org/wiki/MiSeq_SOP"><span
style='font-family:Helvetica'>http://www.mothur.org/wiki/MiSeq_SOP</span></a><span
style='font-family:Helvetica'><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span class=SpellE><i style='mso-bidi-font-style:normal'><span
style='font-size:10.0pt;font-family:Courier;mso-fareast-font-family:"Times New Roman";
mso-bidi-font-family:"Times New Roman";color:#252525;background:white'>Kozich</span></i></span><i
style='mso-bidi-font-style:normal'><span style='font-size:10.0pt;font-family:
Courier;mso-fareast-font-family:"Times New Roman";mso-bidi-font-family:"Times New Roman";
color:#252525;background:white'> JJ, Westcott SL, Baxter NT, Highlander SK, <span
class=SpellE>Schloss</span> PD. (2013): Development of a dual-index sequencing strategy
and <span class=SpellE>curation</span> pipeline for analyzing <span
class=SpellE>amplicon</span> sequence data on the <span class=SpellE>MiSeq</span>
<span class=SpellE>Illumina</span> sequencing platform. <span class=GramE>Applied
and Environmental Microbiology.</span> 79(17)<span class=GramE>:5112</span>-20.</span></i><i
style='mso-bidi-font-style:normal'><span style='font-size:10.0pt;font-family:
Courier;mso-fareast-font-family:"Times New Roman";mso-bidi-font-family:"Times New Roman"'><o:p></o:p></span></i></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal>From the terminal on your instance:<br>
<span style='mso-ascii-font-family:Cambria;mso-fareast-font-family:"Times New Roman";
mso-hansi-font-family:Cambria;mso-bidi-font-family:"Times New Roman";
color:#212121;background:white'>1) <span class=SpellE>ssh</span> 
<span
class=apple-converted-space>&nbsp;studentname@c#.</span><span style='mso-fareast-font-family:
"Times New Roman";mso-bidi-font-family:"Times New Roman"'><a
href="http://programmingforbiology.org/"
data-saferedirecturl="https://www.google.com/url?hl=en&amp;q=http://programmingforbiology.org&amp;source=gmail&amp;ust=1476818972871000&amp;usg=AFQjCNFNRlKfUbiqAbGBhnKi4UOuNNdnpQ"
style='font-variant-ligatures: normal;font-variant-caps: normal;orphans: 2;
text-align:start;widows: 2;-webkit-text-stroke-width: 0px;word-spacing:0px'><span
class=GramE><span style='mso-ascii-font-family:Cambria;mso-hansi-font-family:
Cambria;color:#1155CC;background:white'>programmingforb<wbr>iology.org</span></span></a></span><span
style='mso-ascii-font-family:Cambria;mso-fareast-font-family:"Times New Roman";
mso-hansi-font-family:Cambria;mso-bidi-font-family:"Times New Roman";
color:#500050;background:white'><br>
<span class=im>2) <span class=SpellE>mkdir</span> ~/<span class=SpellE>metagenomic_workshop</span></span><br>
<span class=im>3) cd <span class=SpellE>metagenomic_workshop</span></span><br>
4) <span class=SpellE>cp</span> -R<span class=apple-converted-space>&nbsp;</span><strong
style='font-variant-ligatures: normal;font-variant-caps: normal;orphans: 2;
text-align:start;widows: 2;-webkit-text-stroke-width: 0px;word-spacing:0px'><span
style='font-family:Cambria'>/<span class=SpellE>pfbhome</span>/</span></strong>data/<span
class=SpellE>Metagenomic</span>/16S .</span><br>
5) cd 16S<br>
6) <span class=SpellE>mothur</span><span style='mso-fareast-font-family:"Times New Roman";
mso-bidi-font-family:"Times New Roman"'><o:p></o:p></span></p>

<div style='border:none;border-bottom:solid #AAAAAA 1.0pt;mso-border-bottom-alt:
solid #AAAAAA .75pt;padding:0in 0in 0in 0in'>

<div style='border:none;border-bottom:solid #AAAAAA 1.0pt;mso-border-bottom-alt:
solid #AAAAAA .75pt;padding:0in 0in 0in 0in'>

<div style='border:none;border-bottom:solid #AAAAAA 1.0pt;mso-border-bottom-alt:
solid #AAAAAA .75pt;padding:0in 0in 0in 0in'>

<div style='border:none;border-bottom:solid #AAAAAA 1.0pt;mso-border-bottom-alt:
solid #AAAAAA .75pt;padding:0in 0in 0in 0in'>

<div style='mso-element:para-border-div;border:none;border-bottom:solid #AAAAAA 1.0pt;
mso-border-bottom-alt:solid #AAAAAA .75pt;padding:0in 0in 0in 0in;background:
white'>

<p class=MsoNormal style='margin-top:12.0pt;margin-right:0in;margin-bottom:
3.0pt;margin-left:0in;mso-outline-level:2;background:white;border:none;
mso-border-bottom-alt:solid #AAAAAA .75pt;padding:0in;mso-padding-alt:0in 0in 0in 0in'><span
style='font-size:15.0pt;font-family:Georgia;mso-fareast-font-family:"Times New Roman";
mso-bidi-font-family:"Times New Roman";color:black'>Reducing sequencing and PCR
errors<o:p></o:p></span></p>

</div>

</div>

</div>

</div>

</div>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>In this exercise we are
going to examine the <span class=SpellE>microbiome</span> from the pig
intestinal track from one donor.<span style="mso-spacerun:yes">&nbsp; </span><span
class=GramE>These sequences were generated by 454 downloaded from SRA</span>.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
class=SpellE><i style='mso-bidi-font-style:normal'><span style='font-size:10.0pt;
font-family:Courier;mso-bidi-font-family:Courier;color:black'>Looft</span></i></span><i
style='mso-bidi-font-style:normal'><span style='font-size:10.0pt;font-family:
Courier;mso-bidi-font-family:Courier;color:black'> T, Allen HK, Cantarel BL,
Levine UY, <span class=SpellE>Bayles</span> DO, Alt DP, <span class=SpellE>Henrissat</span>
B, Stanton TB. Bacteria, phages and pigs: the effects of in-feed antibiotics on
<span class=GramE>the<span style="mso-spacerun:yes">&nbsp; </span><span
class=SpellE>microbiome</span></span> at different gut locations. ISME J. 2014
Aug;8(8)<span class=GramE>:1566</span>-76. <span class=SpellE><span
class=GramE>doi</span></span>: 10.1038/ismej.2014.12. <span class=SpellE><span
class=GramE>Epub</span></span><span class=GramE> 2014 Feb 13.</span> PubMed
PMID: 24522263.<o:p></o:p></span></i></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>First lets examine the
sequence data<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>summary.seqs</span></span></span><span style='font-family:Courier'>(<span
class=SpellE>fasta</span>=<span class=SpellE>swine_gut.fasta</span>)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>Now lets trim sequences
based on quality, remove any large, small, <span class=SpellE>ambiguious</span>
or reads with lots of <span class=SpellE>homopolymers</span><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>trim.seqs</span></span></span><span style='font-family:Courier'>(fasta=swine_gut.fasta,qfile=swine_gut.qual,qaverage=25,maxambig=0,maxhomop=8,minlength=200,maxlength=800,processors=4)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>Now we are going to
reduce the number of reads for analysis. Don’t worry we are “keeping” these
sequences for abundance purposes, just not for analysis.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>unique.seqs</span></span></span><span style='font-family:Courier'>(<span
class=SpellE>fasta</span>=<span class=SpellE>swine_gut.trim.fasta</span>)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<div style='border:none;border-bottom:solid #AAAAAA 1.0pt;mso-border-bottom-alt:
solid #AAAAAA .75pt;padding:0in 0in 0in 0in'>

<div style='border:none;border-bottom:solid #AAAAAA 1.0pt;mso-border-bottom-alt:
solid #AAAAAA .75pt;padding:0in 0in 0in 0in'>

<div style='border:none;border-bottom:solid #AAAAAA 1.0pt;mso-border-bottom-alt:
solid #AAAAAA .75pt;padding:0in 0in 0in 0in'>

<div style='border:none;border-bottom:solid #AAAAAA 1.0pt;mso-border-bottom-alt:
solid #AAAAAA .75pt;padding:0in 0in 0in 0in'>

<div style='mso-element:para-border-div;border:none;border-bottom:solid #AAAAAA 1.0pt;
mso-border-bottom-alt:solid #AAAAAA .75pt;padding:0in 0in 0in 0in;background:
white'>

<h2 style='margin-top:12.0pt;margin-right:0in;margin-bottom:3.0pt;margin-left:
0in;background:white;border:none;mso-border-bottom-alt:solid #AAAAAA .75pt;
padding:0in;mso-padding-alt:0in 0in 0in 0in'><span class=mw-headline><span
style='font-size:15.0pt;font-family:Georgia;mso-fareast-font-family:"Times New Roman";
mso-bidi-font-family:"Times New Roman";color:black;font-weight:normal'>Processing
improved sequences</span></span><span style='font-size:15.0pt;font-family:Georgia;
mso-fareast-font-family:"Times New Roman";mso-bidi-font-family:"Times New Roman";
color:black;font-weight:normal'><o:p></o:p></span></h2>

</div>

</div>

</div>

</div>

</div>

<p class=MsoNormal><span style='font-family:Helvetica'>Normally, we can limit
the alignment to the variable region of the 16S gene based on the primers.<span
style="mso-spacerun:yes">&nbsp; </span>First thing we would do is to reduce the
16S reference alignment to this region.<span style="mso-spacerun:yes">&nbsp;
</span>But this step is very long, so we are going to use the precut alignment.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>#<span class=SpellE><span
class=GramE>pcr.seqs</span></span>(<span class=SpellE>fasta</span>=silva.nr_v119.align,
start=1044, end=6500,keepdots=F)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>Next align sequences to
template 16S <span class=SpellE>rRNA</span> gene.<span
style="mso-spacerun:yes">&nbsp; </span>We can remove any sequences that don’t
align to the expected PCR region.<span style="mso-spacerun:yes">&nbsp;&nbsp;
</span>The alignment step can also be very long &#8211; so you should probably
skip this step too.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>#<span class=SpellE><span
class=GramE>align.seqs</span></span>(fasta=swine_gut.trim.unique.fasta,template=silva.nr_v119.pcr.align,flip=t,processors=4)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>Summarize results so far<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>summary.seqs</span></span></span><span style='font-family:Courier'>(<span
class=SpellE>fasta</span>=<span class=SpellE>swine_gut.trim.unique.align</span>,
name=<span class=SpellE>swine_gut.trim.names</span>)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>Next we want to create a
file to keep track of the count of our representative sequences in each
sample.<span style="mso-spacerun:yes">&nbsp; </span>Since we skipped some the <span
class=SpellE>demultiplexing</span> steps, well have to generate the read to
same file (group file).<span style="mso-spacerun:yes">&nbsp; </span>Then create
a count table.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span class=GramE><span style='font-family:Courier'>system</span></span><span
style='font-family:Courier'>(<span class=SpellE>perl</span> make_group_names.pl
<span class=SpellE>swine_gut.trim.names</span>)<o:p></o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>count.seqs</span></span></span><span style='font-family:Courier'>(name=<span
class=SpellE>swine_gut.trim.names</span>, group=<span class=SpellE>swine_gut.trim.groups</span>)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>Now we want to “clean”
up our alignment, determine the sequences that cover most of the PCR region,
cut the alignment to that region and remove those that don’t span the whole
region<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>screen.seqs</span></span></span><span style='font-family:Courier'>(<span
class=SpellE>fasta</span>=<span class=SpellE>swine_gut.trim.unique.align</span>,
count=<span class=SpellE>swine_gut.trim.count_table,start</span>=2,optimize=end,
criteria=85, processors=4)<o:p></o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>filter.seqs</span></span></span><span style='font-family:Courier'>(<span
class=SpellE>fasta</span>=<span class=SpellE>swine_gut.trim.unique.good.align</span>,
vertical=T, trump=., processors=4)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>Now that our alignment
is of high quality, let’s remove redundant sequences and cluster sequences with
only 2 mismatches (these could represent sequencing errors)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>unique.seqs</span></span></span><span style='font-family:Courier'>(<span
class=SpellE>fasta</span>=<span class=SpellE>swine_gut.trim.unique.good.filter.fasta</span>,
count=<span class=SpellE>swine_gut.trim.good.count_table</span>)<o:p></o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>pre.cluster</span></span></span><span style='font-family:Courier'>(fasta=swine_gut.trim.unique.good.filter.unique.fasta,count=swine_gut.trim.unique.good.filter.count_table,diffs=2)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>We want to identify and
remove chimera sequences.<span style="mso-spacerun:yes">&nbsp; </span>This step
is really <span class=GramE>slow,</span> we have already identified these
sequences.<span style="mso-spacerun:yes">&nbsp; </span>So skip this step and
remove the chimeras.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><span
style="mso-spacerun:yes">&nbsp;</span><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>#</span><span
class=SpellE><span class=GramE><span style='font-family:Courier'>chimera.uchime</span></span></span><span
style='font-family:Courier'>(<span class=SpellE>fasta</span>=<span
class=SpellE>swine_gut.trim.unique.good.filter.unique.precluster.fasta</span>,
reference=<span class=SpellE>silva.gold.align,processors</span>=3) <span
class=SpellE>remove.seqs</span>(accnos=swine_gut.trim.unique.good.filter.unique.precluster.ref.uchime.accnos,
fasta=swine_gut.trim.unique.good.filter.unique.precluster.fasta,count=swine_gut.trim.unique.good.filter.unique.precluster.count_table)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<div style='border:none;border-bottom:solid #AAAAAA 1.0pt;mso-border-bottom-alt:
solid #AAAAAA .75pt;padding:0in 0in 0in 0in'>

<div style='border:none;border-bottom:solid #AAAAAA 1.0pt;mso-border-bottom-alt:
solid #AAAAAA .75pt;padding:0in 0in 0in 0in'>

<div style='border:none;border-bottom:solid #AAAAAA 1.0pt;mso-border-bottom-alt:
solid #AAAAAA .75pt;padding:0in 0in 0in 0in'>

<div style='border:none;border-bottom:solid #AAAAAA 1.0pt;mso-border-bottom-alt:
solid #AAAAAA .75pt;padding:0in 0in 0in 0in'>

<div style='mso-element:para-border-div;border:none;border-bottom:solid #AAAAAA 1.0pt;
mso-border-bottom-alt:solid #AAAAAA .75pt;padding:0in 0in 0in 0in;background:
white'>

<h2 style='margin-top:12.0pt;margin-right:0in;margin-bottom:3.0pt;margin-left:
0in;background:white;border:none;mso-border-bottom-alt:solid #AAAAAA .75pt;
padding:0in;mso-padding-alt:0in 0in 0in 0in'><span class=mw-headline><span
style='font-size:15.0pt;font-family:Georgia;mso-fareast-font-family:"Times New Roman";
mso-bidi-font-family:"Times New Roman";color:black;font-weight:normal'>Taxonomic
Classification</span></span><span style='font-size:15.0pt;font-family:Georgia;
mso-fareast-font-family:"Times New Roman";mso-bidi-font-family:"Times New Roman";
color:black;font-weight:normal'><o:p></o:p></span></h2>

</div>

</div>

</div>

</div>

</div>

<p class=MsoNormal><span style='font-family:Helvetica'>#Classify Sequences into
Taxonomic Bins<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>classify.seqs</span></span></span><span style='font-family:Courier'>(fasta=swine_gut.trim.unique.good.filter.unique.precluster.pick.fasta,
template=silva.nr_v119.pcr.align, taxonomy=silva.nr_v119.tax, cutoff=50,
processors=4)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>This command allows you
to remove suspicious sequences, for example if you <span class=SpellE>susptect</span>
Cyanobacteria as representing chloroplast from plant material (which it could
in the gut), you can remove it <o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>remove.lineage</span></span></span><span style='font-family:Courier'>(fasta=swine_gut.trim.unique.good.filter.unique.precluster.pick.fasta,
count=swine_gut.trim.unique.good.filter.unique.precluster.pick.count_table,
taxonomy=swine_gut.trim.unique.good.filter.unique.precluster.pick.nr_v119.wang.taxonomy,taxon=Cyanobacteria)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>#<span class=SpellE><span
class=GramE>mothur</span></span> has a tendency to make long complicated names,
these commands allows you to rename files to shorter names<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span class=GramE><span style='font-family:Courier'>system</span></span><span
style='font-family:Courier'>(<span class=SpellE>cp</span>
swine_gut.trim.unique.good.filter.unique.precluster.pick.pick.fasta <span
class=SpellE>final.fasta</span>)<o:p></o:p></span></p>

<p class=MsoNormal><span class=GramE><span style='font-family:Courier'>system</span></span><span
style='font-family:Courier'>(<span class=SpellE>cp</span>
swine_gut.trim.unique.good.filter.unique.precluster.pick.pick.count_table <span
class=SpellE>final.count_table</span>)<o:p></o:p></span></p>

<p class=MsoNormal><span class=GramE><span style='font-family:Courier'>system</span></span><span
style='font-family:Courier'>(<span class=SpellE>cp</span>
swine_gut.trim.unique.good.filter.unique.precluster.pick.nr_v119.wang.pick.taxonomy
<span class=SpellE>final.taxonomy</span>)<o:p></o:p></span></p>

<div style='border:none;border-bottom:solid #AAAAAA 1.0pt;mso-border-bottom-alt:
solid #AAAAAA .75pt;padding:0in 0in 0in 0in'>

<div style='border:none;border-bottom:solid #AAAAAA 1.0pt;mso-border-bottom-alt:
solid #AAAAAA .75pt;padding:0in 0in 0in 0in'>

<div style='border:none;border-bottom:solid #AAAAAA 1.0pt;mso-border-bottom-alt:
solid #AAAAAA .75pt;padding:0in 0in 0in 0in'>

<div style='border:none;border-bottom:solid #AAAAAA 1.0pt;mso-border-bottom-alt:
solid #AAAAAA .75pt;padding:0in 0in 0in 0in'>

<div style='mso-element:para-border-div;border:none;border-bottom:solid #AAAAAA 1.0pt;
mso-border-bottom-alt:solid #AAAAAA .75pt;padding:0in 0in 0in 0in;background:
white'>

<h2 style='margin-top:12.0pt;margin-right:0in;margin-bottom:3.0pt;margin-left:
0in;background:white;border:none;mso-border-bottom-alt:solid #AAAAAA .75pt;
padding:0in;mso-padding-alt:0in 0in 0in 0in'><span class=mw-headline><span
style='font-size:15.0pt;font-family:Georgia;mso-fareast-font-family:"Times New Roman";
mso-bidi-font-family:"Times New Roman";color:black;font-weight:normal'>Preparing
for OTU analysis</span></span><span style='font-size:15.0pt;font-family:Georgia;
mso-fareast-font-family:"Times New Roman";mso-bidi-font-family:"Times New Roman";
color:black;font-weight:normal'><o:p></o:p></span></h2>

</div>

</div>

</div>

</div>

</div>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>Create a distance matrix
comparing 16S sequences in your samples.<span style="mso-spacerun:yes">&nbsp;
</span>Then cluster sequences into OTU -- we are using the command <span
class=SpellE>cluster.split</span> to do that as it allows us to cluster
sequences according a taxonomic level like order or family<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>dist.seqs</span></span></span><span style='font-family:Courier'>(<span
class=SpellE>fasta</span>=<span class=SpellE>final.fasta,cutoff</span>=0.15,
processors=4)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>#<span class=SpellE><span
class=GramE>cluster.split</span></span>(column=<span class=SpellE>final.dist</span>,
taxonomy=<span class=SpellE>final.taxonomy</span>, count=<span class=SpellE>final.count_table</span>,
<span class=SpellE>splitmethod</span>=classify, <span class=SpellE>taxlevel</span>=4,
processors=4)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>To know the taxonomy for
each of our OTUs, get the consensus taxonomy for each OTU<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>classify.otu</span></span></span><span style='font-family:Courier'>(list=<span
class=SpellE>final.an.unique_list.list</span>, count=<span class=SpellE>final.count_table</span>,
taxonomy=<span class=SpellE>final.taxonomy</span>, label=0.03)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>#Calculate the relative
abundance of taxa<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>phylotype</span></span></span><span style='font-family:Courier'>(taxonomy=<span
class=SpellE>final.taxonomy,count</span>=<span class=SpellE>final.count_table</span>)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-family:Helvetica'>The number of sequences in your count table should
match the number in your OTU list file.<span style="mso-spacerun:yes">&nbsp;
</span>Please continue with the exercise below &#8211; the file <span
class=SpellE>final.fix.count_table</span> is available.<span
style="mso-spacerun:yes">&nbsp; </span><o:p></o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><i style='mso-bidi-font-style:
normal'><span style='font-family:Helvetica'>For Extra Fun:<o:p></o:p></span></i></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><i style='mso-bidi-font-style:
normal'><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></i></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><i style='mso-bidi-font-style:
normal'><span style='font-family:Helvetica'>While some running the final steps
below (some of which are slow) are running try to write your own script to
remove the extra sequences in <span class=SpellE>final.count_table</span> using
the list of sequences in <span class=SpellE>final.an.unique_list.list</span>.<o:p></o:p></span></i></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><i style='mso-bidi-font-style:
normal'><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></i></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><i style='mso-bidi-font-style:
normal'><span style='font-family:Helvetica'>Here are some things to consider:<o:p></o:p></span></i></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><i style='mso-bidi-font-style:
normal'><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></i></b></p>

<p class=MsoListParagraph style='text-indent:-.25in;mso-list:l0 level1 lfo2'><![if !supportLists]><b
style='mso-bidi-font-weight:normal'><i style='mso-bidi-font-style:normal'><span
style='font-family:Helvetica;mso-fareast-font-family:Helvetica;mso-bidi-font-family:
Helvetica'><span style='mso-list:Ignore'>1)<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;
</span></span></span></i></b><![endif]><span class=SpellE><span class=GramE><b
style='mso-bidi-font-weight:normal'><i style='mso-bidi-font-style:normal'><span
style='font-family:Helvetica'>final.an.unique</span></i></b></span><b
style='mso-bidi-font-weight:normal'><i style='mso-bidi-font-style:normal'><span
style='font-family:Helvetica'>_list.list</span></i></b></span><b
style='mso-bidi-font-weight:normal'><i style='mso-bidi-font-style:normal'><span
style='font-family:Helvetica'> is formatted as a label, number of reads and the
reads in each OTU using tab, for example:<o:p></o:p></span></i></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><i style='mso-bidi-font-style:
normal'><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></i></b></p>

<p class=MsoNormal><span class=GramE><b style='mso-bidi-font-weight:normal'><i
style='mso-bidi-font-style:normal'><span style='font-family:Helvetica'>unique</span></i></b></span><b
style='mso-bidi-font-weight:normal'><i style='mso-bidi-font-style:normal'><span
style='font-family:Helvetica'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span></i></b><i
style='mso-bidi-font-style:normal'><span style='font-size:10.0pt;font-family:
Monaco;mso-bidi-font-family:Monaco;color:black'>20551<span style='mso-tab-count:
1'> </span>read1,read2<span style='mso-tab-count:1'> </span>read3,read4<span
style='mso-tab-count:1'> </span>read5,read6,read7</span></i><b
style='mso-bidi-font-weight:normal'><i style='mso-bidi-font-style:normal'><span
style='font-family:Helvetica'><o:p></o:p></span></i></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><i style='mso-bidi-font-style:
normal'><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></i></b></p>

<p class=MsoListParagraphCxSpFirst style='text-indent:-.25in;mso-list:l0 level1 lfo2'><![if !supportLists]><b
style='mso-bidi-font-weight:normal'><i style='mso-bidi-font-style:normal'><span
style='font-family:Helvetica;mso-fareast-font-family:Helvetica;mso-bidi-font-family:
Helvetica'><span style='mso-list:Ignore'>2)<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;
</span></span></span></i></b><![endif]><b style='mso-bidi-font-weight:normal'><i
style='mso-bidi-font-style:normal'><span style='font-family:Helvetica'>You will
have to generate a list of all of the reads in the file and save them into a
look-up array (hash)<o:p></o:p></span></i></b></p>

<p class=MsoListParagraphCxSpLast style='text-indent:-.25in;mso-list:l0 level1 lfo2'><![if !supportLists]><b
style='mso-bidi-font-weight:normal'><i style='mso-bidi-font-style:normal'><span
style='font-family:Helvetica;mso-fareast-font-family:Helvetica;mso-bidi-font-family:
Helvetica'><span style='mso-list:Ignore'>3)<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;
</span></span></span></i></b><![endif]><span class=SpellE><span class=GramE><b
style='mso-bidi-font-weight:normal'><i style='mso-bidi-font-style:normal'><span
style='font-family:Helvetica'>final.count</span></i></b></span><b
style='mso-bidi-font-weight:normal'><i style='mso-bidi-font-style:normal'><span
style='font-family:Helvetica'>_table</span></i></b></span><b style='mso-bidi-font-weight:
normal'><i style='mso-bidi-font-style:normal'><span style='font-family:Helvetica'>
is formatted with the <span class=SpellE>readname</span> and the number of
reads it represents in every samples for example:<o:p></o:p></span></i></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><i style='mso-bidi-font-style:
normal'><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></i></b></p>

<p class=MsoNormal><span class=GramE><b style='mso-bidi-font-weight:normal'><i
style='mso-bidi-font-style:normal'><span style='font-family:Helvetica'>read1</span></i></b></span><b
style='mso-bidi-font-weight:normal'><i style='mso-bidi-font-style:normal'><span
style='font-family:Helvetica'><span style='mso-tab-count:1'> </span>7<span
style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>8<span
style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>6<span
style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>4<span
style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>9<o:p></o:p></span></i></b></p>

<p class=MsoListParagraphCxSpFirst><b style='mso-bidi-font-weight:normal'><i
style='mso-bidi-font-style:normal'><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></i></b></p>

<p class=MsoListParagraphCxSpMiddle><b style='mso-bidi-font-weight:normal'><i
style='mso-bidi-font-style:normal'><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></i></b></p>

<p class=MsoListParagraphCxSpLast style='text-indent:-.25in;mso-list:l0 level1 lfo2'><![if !supportLists]><b
style='mso-bidi-font-weight:normal'><i style='mso-bidi-font-style:normal'><span
style='font-family:Helvetica;mso-fareast-font-family:Helvetica;mso-bidi-font-family:
Helvetica'><span style='mso-list:Ignore'>4)<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;
</span></span></span></i></b><![endif]><b style='mso-bidi-font-weight:normal'><i
style='mso-bidi-font-style:normal'><span style='font-family:Helvetica'>You will
write the first line (this line has the names of all of the samples) and then
each line where the <span class=SpellE>readname</span> was seen in the <span
class=SpellE>final.an.unique_list.list</span><o:p></o:p></span></i></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></b></p>

<div style='border:none;border-bottom:solid #AAAAAA 1.0pt;mso-border-bottom-alt:
solid #AAAAAA .75pt;padding:0in 0in 0in 0in'>

<div style='border:none;border-bottom:solid #AAAAAA 1.0pt;mso-border-bottom-alt:
solid #AAAAAA .75pt;padding:0in 0in 0in 0in'>

<div style='border:none;border-bottom:solid #AAAAAA 1.0pt;mso-border-bottom-alt:
solid #AAAAAA .75pt;padding:0in 0in 0in 0in'>

<div style='border:none;border-bottom:solid #AAAAAA 1.0pt;mso-border-bottom-alt:
solid #AAAAAA .75pt;padding:0in 0in 0in 0in'>

<div style='mso-element:para-border-div;border:none;border-bottom:solid #AAAAAA 1.0pt;
mso-border-bottom-alt:solid #AAAAAA .75pt;padding:0in 0in 0in 0in;background:
white'>

<p class=MsoNormal style='margin-top:12.0pt;margin-right:0in;margin-bottom:
3.0pt;margin-left:0in;mso-outline-level:2;background:white;border:none;
mso-border-bottom-alt:solid #AAAAAA .75pt;padding:0in;mso-padding-alt:0in 0in 0in 0in'><span
style='font-size:15.0pt;font-family:Georgia;mso-fareast-font-family:"Times New Roman";
mso-bidi-font-family:"Times New Roman";color:black'>Analysis<o:p></o:p></span></p>

</div>

</div>

</div>

</div>

</div>

<p class=MsoNormal><span style='font-size:10.0pt;font-family:Times;mso-fareast-font-family:
"Times New Roman";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>#Generates collector's
curves<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>#The <span class=SpellE>make.shared</span>
creates <span class=GramE>a file that represent</span> the number of times that
an OTU is observed in multiple samples<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>make.shared</span></span></span><span style='font-family:Courier'>(list=<span
class=SpellE>final.an.unique_list.list,count</span>=<span class=SpellE>final.fix.count_table</span>,
label=0.03)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>Let's start our analysis
by analyzing the alpha diversity of the samples. First we will generate
collector's curve of the Chao1 richness estimators and the inverse Simpson
diversity index.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>collect.single</span></span></span><span style='font-family:Courier'>(shared=<span
class=SpellE>final.an.unique_list.shared</span>, <span class=SpellE>calc</span>=<span
class=SpellE>chao-invsimpson</span>, <span class=SpellE>freq</span>=100)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica;mso-fareast-font-family:
"Times New Roman";mso-bidi-font-family:"Times New Roman";color:#252525;
background:white'>This command will generate file ending in *<span class=GramE>.<span
class=SpellE>chao</span></span> and *.<span class=SpellE>invsimpson</span> for
each sample, which can be plotted in your favorite graphing software package.
It is important to point out that Chao1 is really a measure of the minimum
richness in a community, not the full richness of the community. Also, there
isn't much to do with these plots since it's based on a single sampling. One
method often used to get around this problem is to look at rarefaction curves
describing the number of OTUs observed as a function of sampling effort.</span><span
style='font-family:Times;mso-fareast-font-family:"Times New Roman";mso-bidi-font-family:
"Times New Roman"'><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>rarefaction.single</span></span></span><span style='font-family:Courier'>(shared=<span
class=SpellE>final.an.unique_list.shared</span>, <span class=SpellE>calc</span>=sobs,
<span class=SpellE>freq</span>=100)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>#Determine the sequence
count per sample<o:p></o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>count.groups</span></span></span><span style='font-family:Courier'>(shared=<span
class=SpellE>final.an.unique_list.shared</span>)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>This will generate files
ending in *<span class=GramE>.rarefaction</span>, which again can be plotted in
your favorite graphing software package. Alas, rarefaction is not a measure of
richness, but a measure of diversity. If you consider two communities with the
same richness, but different evenness then after sampling a large number of
individuals their rarefaction curves will asymptote to the same value. Since
they have different <span class=SpellE>evennesses</span> the shapes of the
curves will differ. Therefore, selecting a number of individuals to cutoff the
rarefaction curve isn't allowing a researcher to compare samples based on
richness, but their diversity. Finally, let's get a table containing the number
of sequences, the sample coverage, the number of observed OTUs, and the Inverse
Simpson diversity estimate using the <span class=SpellE>summary.single</span>
command. To standardize everything, let's randomly select sequences from each
sample 1000 times and calculate the average (note: that if we set subsample=T,
then it would use the size of the smallest library):<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>summary.single</span></span></span><span style='font-family:Courier'>(shared=<span
class=SpellE>final.an.unique_list.shared</span>, <span class=SpellE>calc</span>=<span
class=SpellE>nseqs</span>-coverage-sobs-<span class=SpellE>invsimpson</span>, subsample=2725)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>Subsample the OTUs using
the minimum number of sequences.<span style="mso-spacerun:yes">&nbsp;
</span>Then, let’s determine <span class=GramE>whether<span
style="mso-spacerun:yes">&nbsp; </span>our</span> data can be <span
class=SpellE>partintioned</span> in to separate community types. </span><span
style='font-size:10.0pt;font-family:Times;mso-fareast-font-family:"Times New Roman";
mso-bidi-font-family:"Times New Roman"'><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>sub.sample</span></span></span><span style='font-family:Courier'>(shared=<span
class=SpellE>final.an.unique_list.shared</span>, size=2725)<o:p></o:p></span></p>

<p>Normally, we might want to explore the number of different bacterial
community types that exist in the sample. But in this data, is too small to
determine an accurate measure. </p>

<p class=MsoNormal><span style='font-family:Courier'>#<span class=SpellE><span
class=GramE>get.communitytype</span></span>(shared=final.an.unique_list.0.03.subsample.shared</span><span
style='font-family:Helvetica'>)</span><span style='font-family:Courier'><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica;mso-fareast-font-family:
"Times New Roman";mso-bidi-font-family:"Times New Roman";color:#252525;
background:white'>The <span class=SpellE>unifrac</span>-based metrics are used
to assess the similarity between two communities membership (<span
class=SpellE>unifrac.unweighted</span>) and structure (<span class=SpellE>unifrac.weighted</span>).
We will use these metrics and generate <span class=SpellE>PCoA</span> plots to
compare our samples. There are two beta-diversity metrics that one can use - <span
class=SpellE>unweighted</span> and weighted. We will also have <span
class=SpellE>mothur</span> subsample the trees 1000 times and report the
average.<span style="mso-spacerun:yes">&nbsp; </span></span><span
style='font-family:Helvetica'>In order to do <span class=SpellE>unifract</span>
analysis (beta diversity), we need to determine the distances between <span
class=GramE>sequences,</span> build a <span class=SpellE>phylogentic</span>
tree and then compare samples using <span class=SpellE>unifract</span>.<span
style="mso-spacerun:yes">&nbsp; </span></span><span style='font-family:Helvetica;
mso-fareast-font-family:"Times New Roman";mso-bidi-font-family:"Times New Roman";
color:#252525;background:white'>Principal Coordinates (<span class=SpellE>PCoA</span>)
uses an eigenvector-based approach to represent multidimensional data in as few
<span class=SpellE>dimesnsions</span> as possible. Our data is highly
dimensional (~9 dimensions).<span style="mso-spacerun:yes">&nbsp; </span>Tree
building is a very slow step.<span style="mso-spacerun:yes">&nbsp;
</span>Please use the pre-generated tree.</span><span style='font-family:Times;
mso-fareast-font-family:"Times New Roman";mso-bidi-font-family:"Times New Roman"'><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica;mso-fareast-font-family:
"Times New Roman";mso-bidi-font-family:"Times New Roman";color:#252525;
background:white'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>#<span class=SpellE><span
class=GramE>dist.seqs</span></span>(<span class=SpellE>fasta</span>=<span
class=SpellE>final.fasta,cutoff</span>=0.15, output=<span class=SpellE>lt,processors</span>=4)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'>#<span class=SpellE><span
class=GramE>clearcut</span></span>(<span class=SpellE>phylip</span>=<span
class=SpellE>final.phylip.dist</span>)<o:p></o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>unifrac.unweighted</span></span></span><span style='font-family:Courier'>(tree=<span
class=SpellE>final.phylip.tre</span>, count=<span class=SpellE>final.count_table</span>,
distance=<span class=SpellE>lt</span>, processors=4, random=F, subsample=2725)<o:p></o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>unifrac.weighted</span></span></span><span style='font-family:Courier'>(tree=<span
class=SpellE>final.phylip.tre</span>, count=<span class=SpellE>final.count_table</span>,
distance=<span class=SpellE>lt</span>, processors=4, random=F, subsample=2725)<o:p></o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>pcoa</span></span></span><span style='font-family:Courier'>(<span
class=SpellE>phylip</span>=final.phylip.1.unweighted.ave.dist)<o:p></o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>pcoa</span></span></span><span style='font-family:Courier'>(<span
class=SpellE>phylip</span>=</span><span style='font-family:Courier;mso-bidi-font-family:
Monaco;color:black'>final.phylip.tre1.weighted.ave.dist</span><span
style='font-family:Courier'>)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>We can do the same
analysis using the </span><span style='font-family:Helvetica;mso-fareast-font-family:
"Times New Roman";mso-bidi-font-family:"Times New Roman";color:#252525;
background:white'>similarity of the membership and structure found in the
various samples</span><span style='font-family:Times;mso-fareast-font-family:
"Times New Roman";mso-bidi-font-family:"Times New Roman"'><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>dist.shared</span></span></span><span style='font-family:Courier'>(shared=<span
class=SpellE>final.an.unique_list.shared</span>, <span class=SpellE>calc</span>=<span
class=SpellE>thetayc-jclass</span>, subsample=2725)<o:p></o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>pcoa</span></span></span><span style='font-family:Courier'>(<span
class=SpellE>phylip</span>=final.an.unique_list.thetayc.0.03.lt.dist)<o:p></o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>nmds</span></span></span><span style='font-family:Courier'>(<span
class=SpellE>phylip</span>=final.an.unique_list.thetayc.0.03.lt.dist)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Courier'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'>Finally, no <span
class=SpellE>microbiome</span> study would be complete without a few <span
class=SpellE>heatmaps</span>.<span style="mso-spacerun:yes">&nbsp; </span>We
can visualize those distances as similarities in a <span class=SpellE>heatmap</span>
with the <span class=SpellE>Jaccard</span> and <span class=SpellE>thetayc</span>
coefficients.<span style="mso-spacerun:yes">&nbsp; </span>We can also visualize
relative abundance of each OTU across the 24 samples using the <span
class=SpellE>heatmap.bin</span> command and log2 scaling the relative abundance
values. Because there are so many OTUs, let's just look at the top 50 OTUs.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>heatmap.bin</span></span></span><span style='font-family:Courier'>(shared=<span
class=SpellE>final.an.unique_list.shared</span>, scale=log2, <span
class=SpellE>numotu</span>=50) <o:p></o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>heatmap.sim</span></span></span><span style='font-family:Courier'>(<span
class=SpellE>phylip</span>=final.an.unique_list.thetayc.0.03.lt.dist)<o:p></o:p></span></p>

<p class=MsoNormal><span class=SpellE><span class=GramE><span style='font-family:
Courier'>heatmap.sim</span></span></span><span style='font-family:Courier'>(<span
class=SpellE>phylip</span>=final.an.unique_list.jclass.0.03.lt.dist)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Helvetica'><o:p>&nbsp;</o:p></span></p>

<p><span style='font-size:12.0pt;font-family:Helvetica'>Create a file that you
can import into MEGAN: <o:p></o:p></span></p>

<p><span class=SpellE><span class=GramE><span style='font-size:12.0pt;
font-family:Courier'>make.biom</span></span></span><span style='font-size:12.0pt;
font-family:Courier'>(shared=final.an.unique_list.0.03.subsample.shared,constaxonomy=final.an.unique_list.0.03.cons.taxonomy)
<o:p></o:p></span></p>

<p><span style='font-size:12.0pt;font-family:Helvetica'>Now you want to
download some data to your local machine: <o:p></o:p></span></p>

<p style='margin:0in;margin-bottom:.0001pt'><span style='font-size:12.0pt;
font-family:Helvetica'>1. Open a new tab on your terminal<br>
cd Downloads<br>
<span class=SpellE>mkdir</span> <span class=SpellE>metagenomic_workshops</span><br>
cd <span class=SpellE>metagenomic_workshops</span><br>
<span class=SpellE>scp</span> </span><span style='font-size:12.0pt;font-family:
Helvetica;mso-fareast-font-family:"Times New Roman";color:#212121;background:
white'>
<span style='font-size:12.0pt;font-family:Helvetica'>&nbsp;studentname@c#.</span></span><span
style='font-size:12.0pt;font-family:Helvetica;mso-fareast-font-family:"Times New Roman"'><a
href="http://programmingforbiology.org/"
data-saferedirecturl="https://www.google.com/url?hl=en&amp;q=http://programmingforbiology.org&amp;source=gmail&amp;ust=1476818972871000&amp;usg=AFQjCNFNRlKfUbiqAbGBhnKi4UOuNNdnpQ"
style='font-variant-ligatures: normal;font-variant-caps: normal;orphans: 2;
text-align:start;widows: 2;-webkit-text-stroke-width: 0px;word-spacing:0px'><span
style='color:#1155CC;background:white'>programmingforb<wbr>iology.org</span></a></span><span
class=GramE><span style='font-size:12.0pt;font-family:Helvetica'>:~</span></span><span
style='font-size:12.0pt;font-family:Helvetica'>/metagenomic_workshop/16S/\*.biom
.<br>
<span class=SpellE><span class=GramE>scp</span>
<span
class=apple-converted-space><span style='font-size:12.0pt;font-family:Helvetica'>&nbsp;studentname@c#.</span></span><span
style='font-size:12.0pt;font-family:Helvetica;mso-fareast-font-family:"Times New Roman"'><a
href="http://programmingforbiology.org/"
data-saferedirecturl="https://www.google.com/url?hl=en&amp;q=http://programmingforbiology.org&amp;source=gmail&amp;ust=1476818972871000&amp;usg=AFQjCNFNRlKfUbiqAbGBhnKi4UOuNNdnpQ"
style='font-variant-ligatures: normal;font-variant-caps: normal;orphans: 2;
text-align:start;widows: 2;-webkit-text-stroke-width: 0px;word-spacing:0px'><span
class=GramE><span style='color:#1155CC;background:white'>programmingforb<wbr>iology.org</span></span></a></span><span
style='font-size:12.0pt;font-family:Helvetica'>:/pfbhome/data/Metagenomic/wgs/\*.rma
.<o:p></o:p></span></p>

<p style='margin:0in;margin-bottom:.0001pt'><span class=SpellE><span
class=GramE><span style='font-size:12.0pt;font-family:Helvetica'>scp</span></span></span><span
style='font-size:12.0pt;font-family:Helvetica'> </span>
<span
class=apple-converted-space><span style='font-size:12.0pt;font-family:Helvetica'>&nbsp;studentname@c#.</span></span><span
style='font-size:12.0pt;font-family:Helvetica;mso-fareast-font-family:"Times New Roman"'><a
href="http://programmingforbiology.org/"
data-saferedirecturl="https://www.google.com/url?hl=en&amp;q=http://programmingforbiology.org&amp;source=gmail&amp;ust=1476818972871000&amp;usg=AFQjCNFNRlKfUbiqAbGBhnKi4UOuNNdnpQ"
style='font-variant-ligatures: normal;font-variant-caps: normal;orphans: 2;
text-align:start;widows: 2;-webkit-text-stroke-width: 0px;word-spacing:0px'><span
style='color:#1155CC;background:white'>programmingforb<wbr>iology.org</span></a></span><span
class=GramE><span style='font-size:12.0pt;font-family:Helvetica'>:~</span></span><span
style='font-size:12.0pt;font-family:Helvetica'>/metagenomic_workshop/16S/\*.svg
.<br>
<span class=SpellE><span class=GramE>scp</span></span> </span>
<span
class=apple-converted-space><span style='font-size:12.0pt;font-family:Helvetica'>&nbsp;studentname@c#.</span></span><span
style='font-size:12.0pt;font-family:Helvetica;mso-fareast-font-family:"Times New Roman"'><a
href="http://programmingforbiology.org/"
data-saferedirecturl="https://www.google.com/url?hl=en&amp;q=http://programmingforbiology.org&amp;source=gmail&amp;ust=1476818972871000&amp;usg=AFQjCNFNRlKfUbiqAbGBhnKi4UOuNNdnpQ"
style='font-variant-ligatures: normal;font-variant-caps: normal;orphans: 2;
text-align:start;widows: 2;-webkit-text-stroke-width: 0px;word-spacing:0px'><span
style='color:#1155CC;background:white'>programmingforb<wbr>iology.org</span></a></span><span
class=GramE><span style='font-size:12.0pt;font-family:Helvetica'>:~</span></span><span
style='font-size:12.0pt;font-family:Helvetica'>/metagenomic_workshop/16S/\*.axes\*
.<br>
<span class=SpellE><span class=GramE>scp</span></span> </span>
<span
class=apple-converted-space><span style='font-size:12.0pt;font-family:Helvetica'>&nbsp;studentname@c#.</span></span><span
style='font-size:12.0pt;font-family:Helvetica;mso-fareast-font-family:"Times New Roman"'><a
href="http://programmingforbiology.org/"
data-saferedirecturl="https://www.google.com/url?hl=en&amp;q=http://programmingforbiology.org&amp;source=gmail&amp;ust=1476818972871000&amp;usg=AFQjCNFNRlKfUbiqAbGBhnKi4UOuNNdnpQ"
style='font-variant-ligatures: normal;font-variant-caps: normal;orphans: 2;
text-align:start;widows: 2;-webkit-text-stroke-width: 0px;word-spacing:0px'><span
class=GramE><span style='color:#1155CC;background:white'>programmingforb<wbr>iology.org</span></span></a></span><span
style='font-size:12.0pt;font-family:Helvetica'>:~/metagenomic_workshop/16S/\*.rarefaction
.<o:p></o:p></span></p>

</div>

</body>

</html>
